numpy_groupies/utils_numpy.py

```diff
@@ -219,7 +219,7 @@ def _ravel_group_idx(group_idx, a, axis, size, order, method="ravel"):
     size = []
     for ii, s in enumerate(a.shape):
         if method == "ravel":
-            ii_idx = group_idx_in if ii == axis else np.arange(s)
+            ii_idx = group_idx_in if ii == axis else np.arange(s, like=group_idx_in)
```

Comment:
This is the only change; but requires `numpy >= 1.20.0`. Shall we add that as the min version?

I could also add a version check to switch between the old and new version.

Reply:
The last available numpy for py2 is 1.16.6 as far as I know. This would basically mean to completely drop the weave implementation, which so far is still the best benchmark to compete with for numba. Dropping py2 completely is probably unavoidable at some point, but if there is a way around it, remaining downwards compatible would be nice.